receiver/snmpreceiver/config.go

```diff
@@ -562,13 +571,38 @@ func validateResourceAttributeConfigs(cfg *Config) error {
 		return nil
 	}
 
-	// Make sure each Resource Attribute has either an OID or IndexedValuePrefix
+	// Make sure each Resource Attribute has either an OID or ScalarOID or IndexedValuePrefix
 	for attrName, attrCfg := range resourceAttributes {
-		if attrCfg.OID == "" && attrCfg.IndexedValuePrefix == "" {
-			combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgResourceAttributeNoOIDOrPrefix, attrName))
+
+		hasOID := attrCfg.OID != ""
+		hasScalarOID := attrCfg.ScalarOID != ""
+		hasIVP := attrCfg.IndexedValuePrefix != ""
+
+		switch {
+			case hasOID:
+				if hasScalarOID || hasIVP {
+					combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgMultipleKeysSetOnResourceAttribute, attrName))
```

Comment:
Now that the repo supports a min Go version of 1.20, we're in the process of migrating from `multierr.Append` to `errors.Join`.

Reply:
Also migrated the rest of the receiver (was just config.go)